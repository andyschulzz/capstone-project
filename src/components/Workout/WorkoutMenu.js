import React from 'react'
import * as S from './WorkoutMenu.styles'
import PropTypes from 'prop-types'
import { useHistory } from 'react-router-dom'
import OnOutsiceClick from 'react-outclick'

WorkoutMenu.propTypes = {
  handleWorkoutDelete: PropTypes.func.isRequired,
  handleWorkoutEdit: PropTypes.func.isRequired,
  selectedWorkouts: PropTypes.number,
}

export default function WorkoutMenu({
  title,
  handleWorkoutDelete,
  handleWorkoutEdit,
  toggle,
}) {
  const history = useHistory()
  const ref = React.useRef()
  return (
    <S.Wrapper ref={ref}>
      <p onClick={handleEdit}>
        <S.EditIcon />
        Edit
      </p>
      <p onClick={handleDelete}>
        <S.DeleteIcon />
        Delete
      </p>
      <OnOutsiceClick container={ref} onOutsideClick={click} />
    </S.Wrapper>
  )

  function handleDelete() {
    handleWorkoutDelete(title)
  }

  function handleEdit() {
    handleWorkoutEdit(title)
    history.push('/workouts/edit')
  }

  function click() {
    toggle(false)
  }
}
