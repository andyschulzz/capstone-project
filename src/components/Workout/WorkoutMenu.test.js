import React from 'react'
import { render, cleanup } from '@testing-library/react'
import WorkoutMenu from './WorkoutMenu.js'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)
const handleWorkoutDeleteMock = jest.fn()
const handleWorkoutEditMock = jest.fn()

describe('<WorkoutMenu />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <WorkoutMenu
          title="Test"
          handleWorkoutDelete={handleWorkoutDeleteMock}
          handleWorkoutEdit={handleWorkoutEditMock}
        />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
