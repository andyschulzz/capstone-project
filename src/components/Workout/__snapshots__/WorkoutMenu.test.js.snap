// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkoutMenu /> should render correctly 1`] = `
<DocumentFragment>
  .c1 {
  font-size: 28px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: -10px 20px 5px auto;
  color: hsl(200,100%,50%);
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  right: 0;
  z-index: 1;
  background: #fff;
}

<div
    class="c0"
  >
    <svg
      class="c1"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
